build/config/compiler/compiler.gni

```diff
@@ -52,7 +52,7 @@ declare_args() {
   # warnings with known toolchains. Allow overriding this e.g. for Chromium
   # builds on Linux that could use a different version of the compiler.
   # With GCC, warnings in no-Chromium code are always not treated as errors.
-  treat_warnings_as_errors = true
+  treat_warnings_as_errors = !use_cobalt_customizations
```

Comment:
If we currently are able to treat warnings as errors, we should continue to do that and clean up -Wno flags by cleaning up the code that currently requires them